kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/internal/KmTypes.kt

```diff
@@ -173,8 +173,49 @@ internal fun ImmutableKmTypeParameter.toTypeVariableName(
 
 @KotlinPoetMetadataPreview
 private fun ImmutableKmFlexibleTypeUpperBound.toTypeName(
-  typeParamResolver: ((index: Int) -> TypeName)
+  typeParamResolver: TypeParameterResolver
 ): TypeName {
   // TODO tag typeFlexibilityId somehow?
   return WildcardTypeName.producerOf(type.toTypeName(typeParamResolver))
 }
+internal interface TypeParameterResolver {
+  val parametersMap: Map<Int, TypeVariableName>
+  operator fun get(index: Int): TypeVariableName
+
+  companion object {
+    val EMPTY = object : TypeParameterResolver {
+      override val parametersMap: Map<Int, TypeVariableName> = emptyMap()
+
+      override fun get(index: Int): TypeVariableName = error("No type parameters!")
```

Comment:
If default impl is in place this can be removed, `NoSuchElementException` makes sense for `EMPTY`.

Reply:
ffb271e